package proto;

message peer_data {
  required string ip = 1;
  required int32 age = 3;
  required int64 id = 4;
  required int32 nr_slices = 5;
  required double pos = 6;
  required int32 slice = 7;
}

message pss_message {
  enum Type {
    ANNOUNCE = 0;
    GETVIEW = 1;
    NORMAL = 2;
    RESPONSE = 3;
    LOCAL = 4;
    REQUEST_LOCAL = 5;
    LOADBALANCE = 6;
    LOADBALANCE_LOCAL = 7;
    TERMINATION = 8;
  }

  required Type type = 1;
  required string sender_ip = 2;
  required double sender_pos = 4;
  repeated peer_data view = 5;
}
